core/src/main/java/org/apache/iceberg/DataFilesTable.java

```diff
@@ -57,10 +60,14 @@ public TableScan newScan() {
 
   @Override
   public Schema schema() {
-    Schema schema = new Schema(DataFile.getType(table.spec().partitionType()).fields());
+    List<Types.NestedField> fileFields = Lists.newArrayList();
+    fileFields.addAll(DataFile.getType(table.spec().partitionType()).fields());
+    fileFields.add(ManifestFile.SPEC_ID);
+
+    Schema schema = new Schema(fileFields);
     if (table.spec().fields().size() < 1) {
       // avoid returning an empty struct, which is not always supported. instead, drop the partition field (id 102)
-      return TypeUtil.selectNot(schema, Sets.newHashSet(102));
+      return TypeUtil.selectNot(schema, Sets.newHashSet(DataFile.PARTITION_ID));
```

Comment:
This is a good change, but we should put it in a separate PR.

Reply:
https://github.com/apache/iceberg/pull/1528/files Done